neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -0,0 +1,422 @@
+using Neo.Cryptography.ECC;
+using Neo.IO;
+using Neo.Ledger;
+using Neo.Persistence;
+using Neo.VM;
+using Neo.VM.Types;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Numerics;
+using VMArray = Neo.VM.Types.Array;
+
+namespace Neo.SmartContract.Native.Tokens
+{
+    public class NeoToken : NativeContractBase
+    {
+        public const string ServiceName = "Neo.Native.Tokens.NEO";
+        public static readonly byte[] Script = CreateNativeScript(ServiceName);
+        public static readonly UInt160 ScriptHash = Script.ToScriptHash();
+        public static readonly string[] SupportedStandards = { "NEP-5", "NEP-10" };
+        public const string Name = "NEO";
+        public const string Symbol = "neo";
+        public const int Decimals = 0;
+        public static readonly BigInteger DecimalsFactor = BigInteger.Pow(10, Decimals);
+        public static readonly BigInteger TotalAmount = 100000000 * DecimalsFactor;
+
+        private const byte Prefix_Initialized = 11;
+        private const byte Prefix_Account = 20;
+        private const byte Prefix_Validator = 33;
+        private const byte Prefix_ValidatorsCount = 15;
+
+        internal static bool Main(ApplicationEngine engine)
+        {
+            if (!new UInt160(engine.CurrentContext.ScriptHash).Equals(ScriptHash))
+                return false;
+            string operation = engine.CurrentContext.EvaluationStack.Pop().GetString();
+            VMArray args = (VMArray)engine.CurrentContext.EvaluationStack.Pop();
+            StackItem result;
```

Comment:
Shouldn't we automate this part inside of `NativeContractBase` to have a Unified Entry Point perspective?
Perhaps implement as UnifiedMain(string, args), which has already validated data from stack.